pkg/docker-py3/Dockerfile.tmpl

```diff
@@ -6,15 +6,21 @@ LABEL version="{{ version }}"
 LABEL maintainer="{{ author_email }}"
 
 # Install system requirements. The ORDER of runs is critical. Keep them exactly in this order
-RUN apt update && apt install -y pandoc
-RUN pip install --upgrade pip orderedattrdict tornado
-RUN conda install -c conda-forge -c gramener gramex -y
-RUN apt -y install gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
+RUN apt -qqy --allow-releaseinfo-change update --fix-missing && \
+    apt install -qqy pandoc curl
+RUN apt -qqy install gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
     libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 \
     libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 \
     libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 \
     libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates \
-    fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils
+    fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils \
+    python3-dev default-libmysqlclient-dev build-essential
+RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
+RUN apt update --fix-missing -qqy && \
+    apt install -qqy nodejs --fix-missing && \
+    pip install -q psycopg2-binary==2.9.3 \
+                mysqlclient gramex=="{{ version }}" \
+    gramex setup --all
```

Comment:
The above six lines seem to be replacement for the conda installation. Why do we move to pip instead of conda?

Reply:
yes Those lines are replaced by conda becuase of compatibility issues.
psycorg2 is not found in the library and hence gramex throws errors.

```log
ERROR   10-May 12:15:30 config PORT Exception when importing inputforms.update_model_row
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/pydoc.py", line 347, in safeimport
    module = __import__(path)
  File "/app/inputforms.py", line 11, in <module>
    from psycopg2.extensions import AsIs
ModuleNotFoundError: No module named 'psycopg2'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/gramex/config.py", line 646, in locate
    return _locate(path, forceload)
  File "/opt/conda/lib/python3.7/pydoc.py", line 1607, in locate
    nextmodule = safeimport('.'.join(parts[:n+1]), forceload)
  File "/opt/conda/lib/python3.7/pydoc.py", line 362, in safeimport
    raise ErrorDuringImport(path, sys.exc_info())
pydoc.ErrorDuringImport: problem in inputforms - ModuleNotFoundError: No module named 'psycopg2'
```

mysql client is absolutely necessary to create connections, the existing client does not create connections and throws errors.
Node JS is needed for node building (copying the `node_modules` folder does not work, error for reference )

```log
ERROR   10-May 12:17:40 web Uncaught exception GET /style.scss (172.17.0.1)
HTTPServerRequest(protocol='http', host='localhost:8080', method='GET', uri='/style.scss', version='HTTP/1.1', remote_ip='172.17.0.1')
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/tornado/web.py", line 1592, in _execute
    result = yield result
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.7/site-packages/gramex/handlers/filehandler.py", line 204, in get
    yield self._get_path(self.root / path if self.root.is_dir() else self.root)
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.7/site-packages/gramex/handlers/filehandler.py", line 312, in _get_path
    item = yield item
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "/opt/conda/lib/python3.7/site-packages/gramex/transforms/template.py", line 52, in scss
    result = yield sass2(handler, handler.path)
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/opt/conda/lib/python3.7/site-packages/tornado/gen.py", line 1147, in run
    yielded = self.gen.send(value)
  File "/opt/conda/lib/python3.7/site-packages/gramex/apps/ui/__init__.py", line 144, in sass2
    raise RuntimeError('node-sass compilation failure', proc.stderr)
RuntimeError: ('node-sass compilation failure', '{\n  "status": 1,\n  "file": "/app/style.scss",\n  "line": 71,\n  "column": 22,\n  "message": "wrong number of arguments (4 for 3) for `rgb\'",\n  "formatted": "Error: wrong number of arguments (4 for 3) for `rgb\'\\n        
on line 71 of style.scss\\n        from line 1 of ../root/.config/gramexdata/apps/ui/theme-f6c8d.scss\\n>>   \\"color-high-cell\\": rgb(247, 32, 71, 0.7),\\r\\n   ---------------------^\\n"\n}\n')
```

`gramex setup --all`
This is essential, without it no matter what we do, js are not compiled. scss, sass are not compiled.
According to the documentation also, this is essential step.
Also while we are at it, it will be great to throw in one `npm i --quiet` also.

Reply:
I'll attempt to make the image smaller later.
This time the attempt failed hence I did not commit that part.